### PR TITLE
Bump the Goptuna version up to v0.8.0 with IPOP-CMA-ES and BIPOP-CMA-ES support.

### LINKED ISSUE #1283: Upgrade Goptuna to v0.8.0 which supports IPOP-CMA-ES and BIPOP-CMA-ES.
/kind feature

**Describe the solution you'd like**

I've just released [Goptuna v0.6.0](https://github.com/c-bata/goptuna/releases/tag/v0.6.0). It includes IPOP-CMA-ES and BIPOP-CMA-ES supports. As the benchmark results of two-dimensional [Rastrigin function](https://www.sfu.ca/~ssurjano/rastr.html) shows, these algorithms outperforms on multimodal functions.

rastrigin-function-dim-2-1a14f4610cabf55e51b3b006bfe723dcadef3a57d9eb5c82a09c2c9fbb804c42

* [Auger, A., Hansen, N.: A restart CMA evolution strategy with increasing population size. In: Proceedings of the 2005 IEEE Congress on Evolutionary Computation (CEC’2005), pp. 1769–1776 (2005a)](https://sci2s.ugr.es/sites/default/files/files/TematicWebSites/EAMHCO/contributionsCEC05/auger05ARCMA.pdf)
* [Hansen N. Benchmarking a BI-Population CMA-ES on the BBOB-2009 Function Testbed. In the workshop Proceedings of the Genetic and Evolutionary Computation Conference, GECCO, pages 2389–2395. ACM, 2009.](https://hal.inria.fr/inria-00382093/document)

**TODOs**

* [ ] Add Katib options to enable IPOP-CMA-ES and BIPOP-CMA-ES.
* [ ] Remove the use of deprecated APIs (Some study options are renamed).

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by c-bata**
**What this PR does / why we need it**:
Upgrade goptuna version to v0.8.0 with following changes:

* Add Katib options to enable IPOP-CMA-ES and BIPOP-CMA-ES.
* Remove the use of deprecated APIs (Some study options are renamed without breaking changes).

ScreenShot 2021-04-22 19 17 55

https://github.com/c-bata/katib-goptuna-example/blob/master/experiment.yaml

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1283

**Special notes for your reviewer**:

1. ~~I added changes in the new UI (`pkg/new-ui/v1beta1/frontend/src/app/constants/algorithms-settings.const.ts`) but not checked whether it works.~~
1. Goptuna supports Sobol's Quasi-random Sequence. I have a plan to add Sobol support in the following pull request.

**Release note**:

```release-note
Support IPOP-CMA-ES and BIPOP-CMA-ES, restarting strategies for CMA-ES algorithm.
```

### CODE REVIEW

File: examples/v1beta1/cmaes-example.yaml [hunk 1]
```diff
@@ -12,6 +12,9 @@ spec:
       - Train-accuracy
   algorithm:
     algorithmName: cmaes
+    algorithmSettings:
+      - name: "restart_strategy"
+        value: "ipop"
```
Comment (andreyvelich) [inline examples/v1beta1/cmaes-example.yaml:17]: I think we should also update the Kubeflow docs with the `restart_strategy`: https://www.kubeflow.org/docs/components/katib/experiment/#covariance-matrix-adaptation-evolution-strategy-cma-es.

**Review by andreyvelich — COMMENTED** [review]
Thank you @c-bata, this is very helpful!
/cc @kimwnasptd for the UI review.

File: examples/v1beta1/cmaes-example.yaml (hunk 1 above)
Comment (c-bata) [reply examples/v1beta1/cmaes-example.yaml:17]: I've just opened a draft pull request at https://github.com/kubeflow/website/pull/2642 👍

**Comment by kimwnasptd** [comment]
/lgtm

Thanks for also updating the docs :smile:

File: pkg/new-ui/v1beta1/frontend/src/app/constants/algorithms-settings.const.ts [hunk 2]
```diff
@@ -89,6 +89,12 @@ export const CMAESSettings: AlgorithmSetting[] = [
     value: 0.001,
     type: AlgorithmSettingType.FLOAT,
   },
+  {
+    name: 'restart_strategy',
+    value: 'none',
+    values: ['none', 'ipop', 'bipop'],
+    type: AlgorithmSettingType.STRING,
+  },
```
Comment (c-bata) [inline pkg/new-ui/v1beta1/frontend/src/app/constants/algorithms-settings.const.ts:97]: I checked the new Katib UI.

ScreenShot 2021-04-23 21 15 12

**Comment by andreyvelich** [comment]
Looks great, thanks @c-bata!
/approve

**Pull request merged**